src/plugins/data_view_editor/public/components/data_view_editor_flyout_content.tsx

```diff
@@ -273,9 +302,24 @@ const IndexPatternEditorFlyoutContentComponent = ({
     [http, allowHidden, allSources, type, rollupIndicesCapabilities, searchClient, isLoadingSources]
   );
 
+  // If editData exists, loadSources so that MatchedIndices can be loaded for the Timestampfields
+  useEffect(() => {
+    if (editData) loadSources();
+  }, [editData, loadSources]);
+
+  useEffect(() => {
+    if (editData) reloadMatchedIndices(editData.title);
+  }, [editData, allSources, reloadMatchedIndices]);
+
+  useEffect(() => {
```

Comment:
When editing a data view and turning on `Allow hidden and system indices` first matched indices load but then reset back as if `Allow hidden and system indices` is disabled. 

Probably something wrong with state management here and related to timestamp